python/ray/serve/controller.py

```diff
@@ -76,6 +83,55 @@ class Config:
         arbitrary_types_allowed = True
 
 
+@dataclass
+class MetadataInfo:
+    backends: Dict[BackendTag, BackendInfo]
+    traffic_policies: Dict[EndpointTag, TrafficPolicy]
+
+    def __init__(self):
+        self.backends = dict()
+        self.traffic_policies = dict()
+
+    def get_backend_configs(
+            self, as_dict: bool = False
+    ) -> Dict[BackendTag, Union[BackendConfig, Dict[str, Any]]]:
+        result = {}
+        for tag, info in self.backends.items():
+            config = info.backend_config
+            result[tag] = config.__dict__ if as_dict else config
+        return result
+
+
+@dataclass
+class LifetimeInfo:
+    routers_cache: Dict[NodeId, ActorHandle]
+    routes: Dict[BackendTag, Tuple[EndpointTag, Any]]
```

Comment:
this should be in metadatainfo

Reply:
Hmm, I definitely see the point of moving **routes**, because it has some information about the methods. I'm not 100% sure about `routers_cache` because they seem like more of a running state construct?